File: WebMacroSoftKeyboard/ClientAppFlutter/web_macro_soft_keyboard_client/lib/models/panel.dart
import 'ViewConfigParameter.dart';

class Panel {
  final int id;
  final int width;
  final int height;
  final String pluginName;
  final String symbolicName;
  final int order;
  final List<ViewConfigParameter> configParameters;

  Panel({
    required this.id,
    required this.width,
    required this.height,
    required this.pluginName,
    required this.symbolicName,
    required this.order,
    required this.configParameters,
  });

  Panel.fromJson(Map<String, dynamic> json)
      : id = json["id"],
        width = json["width"],
        height = json["height"],
        pluginName = json["pluginName"],
        symbolicName = json["symbolicName"],
        order = json["order"],
        configParameters = List.castFrom(json["configParameters"]).map<ViewConfigParameter>((jsonConfigParameter) => ViewConfigParameter.fromJson(jsonConfigParameter)).toList();
}

File: MakroBoard.Data/Panel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace MakroBoard.Data
{
    public class Panel
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }

        public int Witdh { get; set; }

        public int Height { get; set; }

        public string PluginName { get; set; }

        public string SymbolicName { get; set; }

        public int Order {get; set; }

        [ForeignKey("Group")]
        public int GroupID  { get; set; }

        [JsonIgnore]
        public Group Group {get; set; }

        [InverseProperty("Panel")]
        public List<ConfigParameterValue> ConfigParameters {get; set;}

    }


}